### ISSUE TITLE
client sending pingresp instead of ping req

### ISSUE THREAD

**Issue by kteza1** [zonyitoo/mqtt-rs#2]
https://github.com/zonyitoo/mqtt-rs/blob/master/examples/pub-client.rs#L120-L124

Isn't it the client that should be sending PINGREQ packets there ? Also, do you have any plans of exposing `keep_alive` to user API while creating a connect packet?

**Comment by zonyitoo** [comment]
Hmm, I took a look at the detail of the standard, I think I was wrong.
http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/errata01/os/mqtt-v3.1.1-errata01-os-complete.html#_Toc442180894

Yes, this `PINGREQ` should be sent only by Client, as defined. But in practice, I found that the server will also send `PINGREQ`, which is why I handle the `PINGREQ` here and send `PINGRESP` back.

For the missing setter for `keep_alive`, I think it is my fault. I have forgot about it. Are you going to open an PR for this? Or I will add it directly.
